core/src/test/java/io/parsingdata/metal/expression/value/BinaryValueExpressionListSemanticsTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.value;
+
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.encoding.Encoding;
+import io.parsingdata.metal.expression.Expression;
+import io.parsingdata.metal.token.Token;
+import io.parsingdata.metal.util.ParameterizedParse;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static io.parsingdata.metal.Shorthand.*;
+import static io.parsingdata.metal.TokenDefinitions.any;
+import static io.parsingdata.metal.util.EncodingFactory.*;
+import static io.parsingdata.metal.util.EnvironmentFactory.stream;
+
+public class BinaryValueExpressionListSemanticsTest extends ParameterizedParse {
+
+    @Parameterized.Parameters(name="{0} ({4})")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+            { "a, a, b, b, last(a+b)", pred2(eq(last(add(ref("a"), ref("b"))))), stream(1, 2, 3, 4, 6), enc(), true },
+            { "a, a, b, b, first(a+b)", pred2(eq(first(add(ref("a"), ref("b"))))), stream(1, 2, 3, 4, 4), enc(), true },
+            { "a, a, a, b, b, last(a+b)", pred3(eq(last(add(ref("a"), ref("b"))))), stream(1, 2, 3, 4, 5, 8), enc(), true },
```

Comment:
`last(add(ref("a"), ref("b")))`, that should not be 8, in this case, right?

- `ref("a")` :arrow_right: `[1,2,3]`
- `ref("b")` :arrow_right: `[4,5]`
- `add(ref("a"), ref("b"))` :arrow_right: `[1+4,2+5,3+ω]` =  `[5,7,ω]`
- `last(add(ref("a"), ref("b")))` :arrow_right: ω ?

Or is it 'left' padded with `ω`, because the `head` of the lists point to `3` and `5`?

- `add(ref("a"), ref("b"))` :arrow_right: `[1+ω,2+4,3+5]` =  `[ω, 6,8]`